blog/_posts/2025-04-04-Traffic-Routing-Demystified-Best-practices-for-AKS-Load-Balancers.md

```diff
@@ -0,0 +1,145 @@
+
+2. **Announce Readiness as `false`**:
+     The application should update its readiness probe to indicate it is no longer ready to serve traffic after the above delay has completed. This allows Kubernetes to stop routing new connections to the pod after the load balancer removes it from the list of active pods.
+
+3. **Gracefully Close Existing Connections**:
+     The application should close all active connections, ensuring that no in-flight requests are dropped.
+
+4. **Exit the Process**:
+     Once all shutdown tasks are complete, the application should terminate its process cleanly.
+
+By implementing these best practices, you can minimize disruptions during pod shutdowns, maintain a seamless user experience, and ensure the reliability of your services in production environments.
+
+## Best Practices for Rolling Updates and Pod Rotation
+![Pod Rotation Timeline](./AKS/blog/assets/images/podrotationbestpractices.png)
+
+While the above works when a pod is being taken down in isolation, it does not cover cases like upgrades and rolling restarts which require coordination between the time the pod goes down and a new one comes up, ready to serve traffic. To optimize pod rotation, add the following best practice to your deployment:
+
+- **Set `minReadySeconds`**:
+    Configure the `minReadySeconds` parameter in your deployment (we recommend around 10 sec) to introduce a delay before Kubernetes is able to mark the pod as "available" (i.e the pod has been ready long enough that the rolling upgrade can move to the next pod - making it different from the "ready" state which implies the application is ready to recieve new connections). This buffer gives the load balancer enough time to register the new pod and start routing traffic to it, while also preventing Kubernetes from deleting the old pod prematurely.
+
+By implementing this strategy, you can achieve smoother rolling updates and maintain a consistent user experience during application changes.
+
+## Pod Distribution Best Practices
+
+Achieving an even distribution of pods across nodes is important for load balancing and resource utilization, especially for pods receiving external traffic via `externalTrafficPolicy=Local`. The diagram below demonstrates an example of uneven pod distribution which leads to imbalanced traffic across pods:
+
+![Pod Distribution](./AKS/blog/assets/images/poddistribution.png)
+
+In this situation, even though the load balancer divides the traffic evenly between nodes, the pods on the node with 2 replicas serve 25% of the traffic each, while the pod in the single replica node serves the full 50% of the total traffic.
+
+Below are some best practices you can follow to evenly distribute pods across your nodes based on your workload needs:
+1. **Use Pod Anti-Affinity**:
```

Comment:
Might be good to link to some sort of reference for more information?  It would be good for all of these recommendations. This currently assumes understanding of these options.